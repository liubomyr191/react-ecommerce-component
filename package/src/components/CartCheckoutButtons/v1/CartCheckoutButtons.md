### CartCheckoutButtons

#### Overview
The `CartCheckoutButtons` displays buttons that allows the user to continue to checkout their cart.

#### Cart checkout buttons will be rendered when the user needs to continue to the checkout step in the cart
```jsx
const onClick = () => {};

  <CartCheckoutButtons
    components={{Button}}
    isDisabled={false}
    onClick={onClick}
  />
```

#### It's possible to customize buttons text
```jsx
const onClick = () => {};

  <CartCheckoutButtons
    buttonText="Custom button text"
    components={{Button}}
    isDisabled={false}
    onClick={onClick}
  />
```
