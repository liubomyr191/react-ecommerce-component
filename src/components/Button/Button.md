A typical example:

```jsx
<Button title="Submit" className="myBtn">Submit</Button>
```
