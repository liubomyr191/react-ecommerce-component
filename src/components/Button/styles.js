export const containerStyles = {
  cursor: "pointer",
  marginBottom: 0,
  marginLeft: 0,
  marginRight: 0,
  marginTop: 0,
  paddingBottom: "0.5rem",
  paddingLeft: "1.5rem",
  paddingRight: "1.5rem",
  paddingTop: "0.5rem",
  textAlign: "center"
};

export const flexContainerStyles = {
  ...containerStyles,
  display: "flex"
};
