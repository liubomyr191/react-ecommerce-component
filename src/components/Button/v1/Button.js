import React, { Component } from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { applyTheme, preventAccidentalDoubleClick } from "helpers";
import { applyThemeWithActionType } from "./helpers";

const paddingFunc = (props) => {
  const { isShortHeight } = props;
  if (isShortHeight) return applyTheme("buttonVerticalPaddingShort");
  return applyTheme("buttonVerticalPadding");
};

const ButtonDiv = styled.div`
  background-color: ${applyThemeWithActionType("buttonBackgroundColor", true)};
  border-color: ${applyThemeWithActionType("buttonBorderColor", true)};
  border-style: solid;
  border-width: 1px;
  border-radius: ${applyTheme("buttonBorderRadius")};
  box-sizing: border-box;
  color: ${applyThemeWithActionType("buttonForegroundColor", false)};
  cursor: pointer;
  display: ${(props) => {
    const { children, fullWidth } = props;
    if (fullWidth) {
      if (typeof children === "string") return "block";
      return "flex";
    }
    if (typeof children === "string") return "inline-block";
    return "inline-flex";
  }};
  margin: 0;
  min-width: ${applyTheme("buttonMinimumWidth")};
  outline: none;
  padding-left: ${applyTheme("buttonHorizontalPadding")};
  padding-right: ${applyTheme("buttonHorizontalPadding")};
  padding-top: ${paddingFunc};
  padding-bottom: ${paddingFunc};
  text-align: center;

  &:hover {
    background-color: ${applyThemeWithActionType("buttonBackgroundColor", true, "hover")};
    border-color: ${applyThemeWithActionType("buttonBorderColor", true, "hover")};
    color: ${applyThemeWithActionType("buttonForegroundColor", false, "hover")};
  }

  &:active {
    background-color: ${applyThemeWithActionType("buttonBackgroundColor", true, "active")};
    border-color: ${applyThemeWithActionType("buttonBorderColor", true, "active")};
    color: ${applyThemeWithActionType("buttonForegroundColor", false, "active")};
  }
`;

class Button extends Component {
  static propTypes = {
    /**
     * The type of action performed by the button
     */
    actionType: PropTypes.oneOf(["danger", "default", "important", "secondary", "secondaryDanger"]),
    /**
     * The contents of the button, such as text, icons, or any combination of React and HTML components
     */
    children: PropTypes.node,
    /**
     * Classes for the button element
     */
    className: PropTypes.string,
    /**
     * Set to `true` to prevent the button from calling `onClick` when clicked
     */
    isDisabled: PropTypes.bool,
    /**
     * Button should take full width
     */
    isFullWidth: PropTypes.bool,
    /**
     * Enable this when you donâ€™t have enough vertical space, such as in table headers
     */
    isShortHeight: PropTypes.bool,
    /**
     * Enable this in rare cases where having a solid or outline button for an action would be cluttered and visually confusing.
     */
    isTextOnly: PropTypes.bool,
    /**
     * Called with no arguments whenever the button is clicked. There is double-click protection,
     * so if the user double-clicks quickly, onClick is called only once.
     */
    onClick: PropTypes.func,
    /**
     * The string for the `title` attribute of the button element
     */
    title: PropTypes.string
  };

  static defaultProps = {
    actionType: "default",
    isDisabled: false,
    isFullWidth: false,
    onClick() {}
  };

  handleClick = preventAccidentalDoubleClick((event) => {
    event.preventDefault();

    const { isDisabled, onClick } = this.props;

    if (!isDisabled) onClick();
  });

  handleKeyPress = (event) => {
    if (event.keyCode === 13) this.handleClick(event);
  };

  render() {
    const { actionType, children, className, isDisabled, isFullWidth, isShortHeight, isTextOnly, title } = this.props;

    const moreButtonDivProps = {};
    if (isDisabled) {
      moreButtonDivProps["aria-disabled"] = "true";
    }

    return (
      <ButtonDiv
        actionType={actionType}
        className={className}
        isDisabled={isDisabled}
        fullWidth={isFullWidth}
        onClick={this.handleClick}
        onKeyPress={this.handleKeyPress}
        role="button"
        isShortHeight={isShortHeight}
        isTextOnly={isTextOnly}
        tabIndex={0}
        title={title}
        {...moreButtonDivProps}
      >
        {children}
      </ButtonDiv>
    );
  }
}

export default Button;
