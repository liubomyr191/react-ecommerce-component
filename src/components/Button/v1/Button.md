The `Button` component can render with four different possible appearances, which are chosen based on the `actionType` prop value. You should choose the appropriate appearance based on what type of action will happen if you click it.

The actual colors of the appearance are defined by your theme. The examples here use our default theme. The theme properties used by this component are:

- `rui_buttonBorderRadius`
- `rui_buttonBorderColor_danger`
- `rui_buttonBorderColor_default`
- `rui_buttonBorderColor_important`
- `rui_buttonBorderColor_secondary`
- `rui_buttonBackgroundColor_danger`
- `rui_buttonBackgroundColor_default`
- `rui_buttonBackgroundColor_important`
- `rui_buttonBackgroundColor_secondary`
- `rui_buttonForegroundColor_danger`
- `rui_buttonForegroundColor_default`
- `rui_buttonForegroundColor_important`
- `rui_buttonForegroundColor_secondary`
- `rui_buttonPadding`

#### Default Button

The default button is the most common button and should be your first choice.

```jsx
<div style={{ display: "flex" }}>
  <div style={{ marginRight: "1rem" }}>
    <Button title="Default" className="myBtn">Default</Button>
  </div>
  <div style={{ marginRight: "1rem" }}>
    <Button title="Default" className="myBtn" disabled>Default Disabled</Button>
  </div>
</div>
```

#### Secondary Button

For secondary actions or actions that are not required. For example, a Cancel button.

```jsx
<div style={{ display: "flex" }}>
  <div style={{ marginRight: "1rem" }}>
    <Button title="Secondary" className="myBtn" actionType="secondary">Secondary</Button>
  </div>
  <div style={{ marginRight: "1rem" }}>
    <Button title="Secondary" className="myBtn" actionType="secondary" disabled>Secondary Disabled</Button>
  </div>
</div>
```

#### Important Button

```jsx
<div style={{ display: "flex" }}>
  <div style={{ marginRight: "1rem" }}>
    <Button title="Important" className="myBtn" actionType="important">Important</Button>
  </div>
  <div style={{ marginRight: "1rem" }}>
    <Button title="Important" className="myBtn" actionType="important" disabled>Important Disabled</Button>
  </div>
</div>
```

#### Danger Button

```jsx
<div style={{ display: "flex" }}>
  <div style={{ marginRight: "1rem" }}>
    <Button title="Danger" className="myBtn" actionType="danger">Danger</Button>
  </div>
  <div style={{ marginRight: "1rem" }}>
    <Button title="Danger" className="myBtn" actionType="danger" disabled>Danger Disabled</Button>
  </div>
</div>
```

#### Full Width Button

To make any button take up the full width of its container, add `fullWidth` prop.

```jsx
const divStyles = {
  borderColor: "#888888",
  borderStyle: "solid",
  borderWidth: 1,
  display: "flex",
  flexDirection: "column",
  height: 100,
  justifyContent: "space-around",
  marginLeft: "auto",
  marginRight: "auto",
  padding: 10,
  width: 500
};

<div style={divStyles}>
  <Button className="myBtn" fullWidth>Full Width Default</Button>
  <Button className="myBtn" actionType="danger" fullWidth>Full Width Danger</Button>
</div>
```
