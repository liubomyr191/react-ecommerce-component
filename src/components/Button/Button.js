import React, { Component } from "react";
import PropTypes from "prop-types";
import preventAccidentalDoubleClick from "../../helpers/preventAccidentalDoubleClick";
import { containerStyles, fullWidthContainerStyles, fullWidthTextContainerStyles } from "./styles";

class Button extends Component {
  static propTypes = {
    /**
     * The type of action performed by the button
     */
    actionType: PropTypes.oneOf(["danger", "default", "important", "secondary"]),
    /**
     * The contents of the button, such as text, icons, or any combination of React and HTML components
     */
    children: PropTypes.node,
    /**
     * Classes for the button element
     */
    className: PropTypes.string,
    /**
     * Set to `true` to prevent the button from calling `onClick` when clicked
     */
    disabled: PropTypes.bool,
    /**
     * Button should take full width
     */
    fullWidth: PropTypes.bool,
    /**
     * Called with no arguments whenever the button is clicked. There is double-click protection,
     * so if the user double-clicks quickly, onClick is called only once.
     */
    onClick: PropTypes.func,
    /**
     * The string for the `title` attribute of the button element
     */
    title: PropTypes.string
  };

  static defaultProps = {
    actionType: "default",
    disabled: false,
    fullWidth: false,
    onClick() {}
  };

  handleClick = preventAccidentalDoubleClick((event) => {
    event.preventDefault();

    const { disabled, onClick } = this.props;

    if (!disabled) {
      onClick();
    }
  });

  handleKeyPress = (event) => {
    if (event.keyCode === 13) this.handleClick(event);
  };

  render() {
    const { children, className, disabled, fullWidth, title, actionType } = this.props;

    // Build className
    const finalClassName = `rui--btn rui--btn-${actionType} ${disabled ? "rui--btn-disabled " : ""}${className}`.trim();

    // Build style
    let style;
    if (fullWidth) {
      if (typeof children === "string") {
        style = fullWidthTextContainerStyles;
      } else {
        style = fullWidthContainerStyles;
      }
    } else {
      style = containerStyles;
    }

    return (
      <div
        className={finalClassName}
        onClick={this.handleClick}
        onKeyPress={this.handleKeyPress}
        role="button"
        style={style}
        tabIndex={0}
        title={title}
      >
        {children}
      </div>
    );
  }
}

export default Button;
