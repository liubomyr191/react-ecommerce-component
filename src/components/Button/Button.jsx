import React, { Component } from "react";
import PropTypes from "prop-types";
import preventAccidentalDoubleClick from "../../helpers/preventAccidentalDoubleClick";
import { containerStyles, flexContainerStyles } from "./styles";

class Button extends Component {
  static propTypes = {
    /**
     * The contents of the button, such as text, icons, or any combination of React and HTML components
     */
    children: PropTypes.node,
    /**
     * Classes for the button element
     */
    className: PropTypes.string,
    /**
     * Set to `true` to prevent the button from calling `onClick` when clicked
     */
    disabled: PropTypes.bool,
    /**
     * Called with no arguments whenever the button is clicked. There is double-click protection,
     * so if the user double-clicks quickly, onClick is called only once.
     */
    onClick: PropTypes.func,
    /**
     * The string for the `title` attribute of the button element
     */
    title: PropTypes.string
  };

  static defaultProps = {
    disabled: false,
    onClick() {}
  };

  onClick = preventAccidentalDoubleClick((event) => {
    const { disabled, onClick } = this.props;

    if (!disabled) {
      onClick(event);
    }
  });

  render() {
    const { children, className, disabled, title } = this.props;

    const finalClassName = `btn-default ${disabled ? "btn-disabled " : ""}${className}`.trim();
    const style = typeof children === "string" ? containerStyles : flexContainerStyles;

    return (
      <div className={finalClassName} onClick={this.onClick} style={style} title={title}>
        {children}
      </div>
    );
  }
}

export default Button;
