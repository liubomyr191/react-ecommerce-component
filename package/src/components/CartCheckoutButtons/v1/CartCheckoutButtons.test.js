import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import Button from "../../Button/v1";
import CartCheckoutButtons from "./CartCheckoutButtons";

test("default cart checkout button", () => {
  const onClick = () => {};
  const isDisabled = false;

  const component = renderer.create((
    <CartCheckoutButtons
      components={{ Button }}
      isDisabled={isDisabled}
      onClick={onClick}
    />
  ));

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("cart checkout button with custom text", () => {
  const onClick = () => {};
  const isDisabled = false;

  const component = renderer.create((
    <CartCheckoutButtons
      buttonText="Custom button text"
      components={{ Button }}
      isDisabled={isDisabled}
      onClick={onClick}
    />
  ));

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});


test("cart checkout button onClick", () => {
  const isDisabled = false;
  const testClick = jest.fn();

  const component = shallow((
    <CartCheckoutButtons
      buttonText="Custom button text"
      components={{ Button }}
      isDisabled={isDisabled}
      onClick={testClick}
    />
  ));

  component.simulate("click");
  expect(testClick).toHaveBeenCalled();
});
