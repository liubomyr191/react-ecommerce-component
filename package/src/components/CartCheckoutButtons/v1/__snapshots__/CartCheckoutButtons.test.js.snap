// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cart checkout button with custom text 1`] = `
.c2 {
  width: 1.125em;
  height: 1.125em;
}

.c0 {
  -webkit-font-smoothing: antialiased;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #1999dd;
  border-color: #1999dd;
  border-style: solid;
  border-width: 1px;
  border-radius: 2px;
  box-sizing: border-box;
  color: #ffffff;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-family: "Source Sans Pro","Helvetica Neue",Helvetica,sans-serif;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 0;
  min-width: 100px;
  outline: none;
  padding-left: 20px;
  padding-right: 20px;
  padding-top: 10px;
  padding-bottom: 10px;
  position: relative;
  text-align: center;
}

.c0:hover {
  background-color: #25b0f9;
  border-color: #25b0f9;
  color: #ffffff;
}

.c0:active {
  background-color: #057dbe;
  border-color: #057dbe;
  color: #ffffff;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  overflow: hidden;
  padding-left: 20px;
  -webkit-transition: width 0.2s ease-out 0s,padding-left 0.2s ease-out 0s,opacity 0.2s ease-out 0.2s;
  transition: width 0.2s ease-out 0s,padding-left 0.2s ease-out 0s,opacity 0.2s ease-out 0.2s;
}

.c1 svg path,
.c1 svg rect {
  fill: #ffffff;
}

<div
  className="c0"
  onClick={[Function]}
  onKeyPress={[Function]}
  role="button"
  tabIndex={0}
  title={undefined}
>
  <div>
    Custom button text
  </div>
  <div
    className="c1"
    style={
      Object {
        "opacity": 0,
        "paddingLeft": 0,
        "width": 0,
      }
    }
  >
    <svg
      className="c2"
      height="40px"
      style={
        Object {
          "enableBackground": "new 0 0 50 50",
        }
      }
      version="1.1"
      viewBox="0 0 50 50"
      width="40px"
      x="0px"
      y="0px"
    >
      <path
        d="M43.935,25.145c0-10.318-8.364-18.683-18.683-18.683c-10.318,0-18.683,8.365-18.683,18.683h4.068c0-8.071,6.543-14.615,14.615-14.615c8.072,0,14.615,6.543,14.615,14.615H43.935z"
        fill="#000"
      >
        <animateTransform
          attributeName="transform"
          attributeType="xml"
          dur="1s"
          from="0 25 25"
          repeatCount="indefinite"
          to="360 25 25"
          type="rotate"
        />
      </path>
    </svg>
  </div>
</div>
`;

exports[`default cart checkout button 1`] = `
.c2 {
  width: 1.125em;
  height: 1.125em;
}

.c0 {
  -webkit-font-smoothing: antialiased;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #1999dd;
  border-color: #1999dd;
  border-style: solid;
  border-width: 1px;
  border-radius: 2px;
  box-sizing: border-box;
  color: #ffffff;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-family: "Source Sans Pro","Helvetica Neue",Helvetica,sans-serif;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 0;
  min-width: 100px;
  outline: none;
  padding-left: 20px;
  padding-right: 20px;
  padding-top: 10px;
  padding-bottom: 10px;
  position: relative;
  text-align: center;
}

.c0:hover {
  background-color: #25b0f9;
  border-color: #25b0f9;
  color: #ffffff;
}

.c0:active {
  background-color: #057dbe;
  border-color: #057dbe;
  color: #ffffff;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  overflow: hidden;
  padding-left: 20px;
  -webkit-transition: width 0.2s ease-out 0s,padding-left 0.2s ease-out 0s,opacity 0.2s ease-out 0.2s;
  transition: width 0.2s ease-out 0s,padding-left 0.2s ease-out 0s,opacity 0.2s ease-out 0.2s;
}

.c1 svg path,
.c1 svg rect {
  fill: #ffffff;
}

<div
  className="c0"
  onClick={[Function]}
  onKeyPress={[Function]}
  role="button"
  tabIndex={0}
  title={undefined}
>
  <div>
    Checkout
  </div>
  <div
    className="c1"
    style={
      Object {
        "opacity": 0,
        "paddingLeft": 0,
        "width": 0,
      }
    }
  >
    <svg
      className="c2"
      height="40px"
      style={
        Object {
          "enableBackground": "new 0 0 50 50",
        }
      }
      version="1.1"
      viewBox="0 0 50 50"
      width="40px"
      x="0px"
      y="0px"
    >
      <path
        d="M43.935,25.145c0-10.318-8.364-18.683-18.683-18.683c-10.318,0-18.683,8.365-18.683,18.683h4.068c0-8.071,6.543-14.615,14.615-14.615c8.072,0,14.615,6.543,14.615,14.615H43.935z"
        fill="#000"
      >
        <animateTransform
          attributeName="transform"
          attributeType="xml"
          dur="1s"
          from="0 25 25"
          repeatCount="indefinite"
          to="360 25 25"
          type="rotate"
        />
      </path>
    </svg>
  </div>
</div>
`;
